p256/src/arithmetic/scalar.rs

```diff
@@ -348,6 +342,72 @@ impl FromUintUnchecked for Scalar {
     }
 }
 
+impl Invert for Scalar {
+    type Output = CtOption<Self>;
+
+    fn invert(&self) -> CtOption<Self> {
+        self.invert()
+    }
+
+    /// Fast variable-time inversion using Stein's algorithm.
+    ///
+    /// Returns none if the scalar is zero.
+    ///
+    /// <https://link.springer.com/article/10.1007/s13389-016-0135-4>
+    ///
+    /// ⚠️ WARNING!
+    ///
+    /// This method should not be used with (unblinded) secret scalars, as its
+    /// variable-time operation can potentially leak secrets through
+    /// sidechannels.
+    #[allow(non_snake_case)]
+    fn invert_vartime(&self) -> CtOption<Self> {
```

Comment:
@fjarri FYI, there's now a dedicated method for variable-time inversions which is used by the generic ECDSA implementation.

This can probably be adapted to work with `k256`?  I had previously tried to make it fully generic, but due to the use of `Self(MODULUS)` below it's relying on support in the underlying field implementation.

I tried porting it to `k256` and it seemed to work, although ECDSA verification performance was only up by ~3%.